plugins/auto-sizes/tests/test-auto-sizes.php

```diff
@@ -33,8 +33,11 @@ public function get_image_tag( int $attachment_id ): string {
 	}
 
 	public function test_hooks(): void {
-		$this->assertSame( 10, has_filter( 'wp_get_attachment_image_attributes', 'auto_sizes_update_image_attributes' ) );
-		$this->assertSame( 10, has_filter( 'wp_content_img_tag', 'auto_sizes_update_content_img_tag' ) );
+		// Skip loading plugin filters if the WordPress core is already loaded auto sizes.
+		if ( ! function_exists( 'wp_sizes_attribute_includes_valid_auto' ) ) {
+			$this->assertSame( 10, has_filter( 'wp_get_attachment_image_attributes', 'auto_sizes_update_image_attributes' ) );
+			$this->assertSame( 10, has_filter( 'wp_content_img_tag', 'auto_sizes_update_content_img_tag' ) );
+		}
```

Comment:
```suggestion
		$this->assertSame( function_exists( 'wp_sizes_attribute_includes_valid_auto' ) ? false : 10, has_filter( 'wp_get_attachment_image_attributes', 'auto_sizes_update_image_attributes' ) );
		$this->assertSame( function_exists( 'wp_sizes_attribute_includes_valid_auto' ) ? false : 10, has_filter( 'wp_content_img_tag', 'auto_sizes_update_content_img_tag' ) );
```

Reply:
I think we can remove these assertions altogether. We have tests that call the functions that those filters affect, so those tests would already fail if our filter callbacks were not hooked to those functions.

We could also remove the `wp_head` hook test if we rewrote `test_auto_sizes_render_generator` to call `wp_head()` and check that our meta tags were included.

Reply:
Thanks! Addressed in https://github.com/WordPress/performance/pull/1547/commits/a7410d292a974095402b1b5e160609a7ceda8f1d